lib/krikri/parser.rb

```diff
@@ -103,5 +103,40 @@ def get_child_nodes(name)
         raise NotImplementedError, "Can't access property #{name}"
       end
     end
+
+    class ValueArray < Array
+      def values
+        map(&:value)
+      end
+
+      def field(*args)
+        result = self
+        args.each do |name|
+          result = result.get_field(name)
+        end
+        result
+      end
+
+      def select(*)
+        self.class.new(super)
+      end
+
+      def match_attribute(name, other)
```

Comment:
That should absolutely work.  They should also be chainable with field selectors as in `uri record.field('mods:location', 'mods:url').match_attribute(:usage, 'primary').field('mods:someField).match_attribute(:access, 'object in context')`